src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,50 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.UserNotOkException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_AGE = 18;
+    private static final int MIN_CHARS = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        try {
+            checkingUser(user);
+            storageDao.add(user);
+            return storageDao.get(user.getLogin());
+        } catch (UserNotOkException e) {
+            throw new RuntimeException(e);
+        }
     }
+
+    private boolean checkingUser(User user)
```

Comment:
The method name `checkingUser` does not follow Java naming conventions. Method names should typically be verbs or verb phrases. Consider renaming it to something like `validateUser` or `checkUserValidity` to better reflect the action it performs.